SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -332,24 +332,50 @@ QueryConstraint InterestFactory::CreateCheckoutRadiusConstraints() const
 	DefaultCheckoutRadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ DefaultCheckoutRadiusMeters };
 	CheckoutRadiusConstraints.OrConstraint.Add(DefaultCheckoutRadiusConstraint);
 
-	// For every interest distance that we still want, add a constraint with the distance for the actor type and all of its derived types.
-	for (const auto& InterestDistanceSquared: ClientInterestDistancesSquared)
+	TMap<float, TArray<UClass*>> DistanceToActorTypes;
+
+	// For every interest distance that we still want, build a map from radius to list of actor types that match that radius.
+	for (const auto& InterestDistanceSquared : ClientInterestDistancesSquared)
+	{
+		const float CheckoutRadiusMeters = FMath::Sqrt(InterestDistanceSquared.Value / (100.0f * 100.0f));
+
+		check(InterestDistanceSquared.Key);
+		if (DistanceToActorTypes.Contains(CheckoutRadiusMeters))
+		{
+			TArray<UClass*> ActorTypes = DistanceToActorTypes[CheckoutRadiusMeters];
```

Comment:
This won't modify the array already in the map, just a temporary new one, that'll get trashed

Reply:
that's sad- what is the correct way of achieving this then? Why does this make a copy?

Reply:
fixed